### DISCUSSION TITLE
Agent Based Model of Cetacean Acoustics

### DISCUSSION THREAD

**Discussion by dolphins4all** [org-arl/UnderwaterAcoustics.jl#50] [General]
### DolphinGames

The project goal is to build an interactive simulation of dolphins and their responses to disturbances, specifically noise changes such as shipping traffic, development, storms, etc.  The simulation area is roughly 2,000 km^2.  The current approach is to use an Agent Based Model to simulate dolphin movements and interactions. At each model step, dolphins may whistle or echolocate and this may be heard by other dolphins (affecting behavior) or hydrophones (affecting our abilities to detect activity). Disturbances and other changes are added once there is some semblance of reality (hopefully) in the basic community behavior.

#### Phase I - simple framework setup (Agents.jl); no social interaction between dolphins, no acoustics
The environment is a continuous grid containing dolphins and an array of several stationary hydrophones. Dolphins swim about in random directions at a specified velocity. At each time step, they see if they are within a detection radius of a hydrophone. If they are, the hydrophone location changes color.  The hydrophones keep track of detections.

#### Phase II - basic integration with UnderwaterAcoustics.jl

Instead of a static number detection range, dolphins emit a whistle at a specified frequency (pure tone, ping-like). UnderwaterAcoustic package calculates arrival times and levels of whistles (RaySolver?).  The hydrophones record the whistles if they are above a certain threshold (i.e. within the range of the dolphin, given hydrophone sensitivty).  The hydrophone location changes color for that time step if they detect a dolphin and tracks statistics.  Hydrophones create spectograms of what the detect from the dolphin population to assist with probabilities of localizations.

#### Phase III - more sophisticated sound models for communication and noise

Background noise to is added to model space.  Vessel noise is added as a parameter, as well as other sound sources. Echolocation, dolphin behavior, other species...TBD.

**Comment by mchitre** [comment]
Interesting work.

What is the bathymetry of the area you plan to model? Deep or shallow? Do you know the exact bathymetry or do you plan to use a constant depth? Does the water temperature vary with depth, or can you assume iso-velocity sound speed profile? The choice of model depends on the answers to some of these questions, and also on the fidelity-speed balance you desire.

Do you know the ambient noise statistics in the area? This is needed to determine detectability. The signal processing at the hydrophone also plays a big role in detection; do you know how you'll be deciding if a received signal is detectable or not?

**Reply by dolphins4all** [reply]
Bathymetry is near flat and shallow (model area is a shelf region) and varies from 22-45 meters.    Exact bathymetry is probably available for this area, if needed.  Bottom type is mostly sand/shells, some mud.

I *think* I can assume an iso-velocity sound profile. In the winter season, it's pretty well mixed; however, summer may be different.  I'm still new to a lot of this and  I'll check into it.

Yes, I have a couple years of ambient noise statistics for the area (and the raw data if I need to redo anything). I also have all of the hydrophone specs.  No, I haven't fully determined what the appropriate way to decide if a received signal is detectable of not.  I had assumed modelling a noise level and filtering for levels over some threshold.

I have some playback experiments that were performed a few years back that might be useful, although at lower frequencies than I am interested in.

I'm still exploring what could be done with a model like this and what level of complexity is appropriate for the various questions.

**Reply by mchitre** [reply]
I'd recommend you start with a flat bathymetry and iso-velocity profile then. With that, [`PekerisRayModel`](https://org-arl.github.io/UnderwaterAcoustics.jl/stable/pm_pekeris.html#PekerisRayModel) should suffice. It is a 2.5D model and very fast, so easily usable for this application, even if you run every signal through it in your simulation.

**Reply by dolphins4all** [reply]
Great...thank you for the suggestion and I'll keep you apprised of the development.

**Reply by dolphins4all** [reply]
It turns out the iso-velocity assumption is only applicable during fall/winter seasons.  We have a significant thermocline in the summer months.  Would the RaySolver model be a better choice for summer months?  I understand it is still in beta testing.  

There are high resolution ocean models for the area and we will eventually do some ping and playback testing, if that helps with model suggestions.

**Reply by mchitre** [reply]
Yes, when you have a non-iso-SVP, you'll need to use either `RaySolver` or `BELLHOP`. `RaySolver` is still experimental, but for most cases, it produces fairly good results. It's your only choice if you need differentiability. `BELLHOP` is more well established, and might be a good choice if you don't care about differentiability.

P.S. You only need to care about differentiability if you plan to use gradient descent or related methods for fine tuning the models based on measurements.

**Reply by dolphins4all** [reply]
Thanks...the PS answered my next question :) 

I'll try both models seems it _seems_ straightforward to swap the models in this package.  I'm not quite sure how much accuracy I really need yet.
